### PR TITLE
make DebugDirectoryBuilder.AddCodeViewEntry public

### LINKED ISSUE #21458: Make AddCodeViewEntry(..., int age) public
Add an overload of AddCodeViewEntry API for building CodeView Debug Directory entry with specified Age != 1.

```C#
namespace System.Reflection.PortableExecutable
{
  class DebugDirectoryBuilder
  {
        /// <summary>
        /// Adds a CodeView entry.
        /// </summary>
        /// <param name="pdbPath">Path to the PDB. Shall not be empty.</param>
        /// <param name="pdbContentId">Unique id of the PDB content.</param>
        /// <param name="portablePdbVersion">Version of Portable PDB format (e.g. 0x0100 for 1.0), or 0 if the PDB is not portable.</param>
        /// <param name="age">Age (iteration) of the PDB. Shall be 1 for Portable PDBs.</param>
        /// <exception cref="ArgumentNullException"><paramref name="pdbPath"/> is null.</exception>
        /// <exception cref="ArgumentException"><paramref name="pdbPath"/> contains NUL character.</exception>
        /// <exception cref="ArgumentOutOfRangeException"><paramref name="age"/> is less than 1.</exception>
        /// <exception cref="ArgumentOutOfRangeException"><paramref name="portablePdbVersion"/> is smaller than 0x0100.</exception>
        public void AddCodeViewEntry(
            string pdbPath,
            BlobContentId pdbContentId,
            ushort portablePdbVersion,
            int age)
  }
}
```

### PR DESCRIPTION
**Pull request by hrrrrustic**
close #21458
Just changed internal to public :D
Looks like there is no reason to make one more method just with another access modifier

### CODE REVIEW

**Comment by hrrrrustic** [comment]
@tommcdon any comments? :D

**Comment by hoyosjs** [comment]
@tommcdon do you know who owns SRM?

**Comment by hoyosjs** [comment]
@tmat

**Review by tmat — APPROVED** [review]

File: src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/DebugDirectory/DebugDirectoryBuilder.cs [hunk 1]
```diff
@@ -97,7 +97,7 @@ public void AddEntry<TData>(DebugDirectoryEntryType type, uint version, uint sta
         /// <exception cref="ArgumentException"><paramref name="pdbPath"/> contains NUL character.</exception>
         /// <exception cref="ArgumentOutOfRangeException"><paramref name="age"/> is less than 1.</exception>
         /// <exception cref="ArgumentOutOfRangeException"><paramref name="portablePdbVersion"/> is smaller than 0x0100.</exception>
-        internal void AddCodeViewEntry(
+        public void AddCodeViewEntry(
```
Comment (ahsonkhan) [inline src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/DebugDirectory/DebugDirectoryBuilder.cs:100]: Do we already have sufficient tests for this?

@carlossanlop how come it didn't get auto-labelled as new API needing docs?

Comment (ahsonkhan) [reply src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/DebugDirectory/DebugDirectoryBuilder.cs:100]: @tmat - thoughts on testing?

**Pull request merged**